ansible/roles/installation/nuc-pre-check/tasks/main.yml

```diff
@@ -0,0 +1,23 @@
+# Copyright 2023 Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+---
+- name: Check if Intel Turbo Boost is enabled
+  shell: cat /sys/devices/system/cpu/intel_pstate/no_turbo
```

Comment:
Is this file always available?
Have you checked if you can use an inbuild function like this? https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_lookup.html

Reply:
Try to use ansible calls whenever possible. Use shell and cmd only when there isnt one available in ansible

Reply:
Applied you suggestions :)